iot_packet_verifier/src/verifier.rs

```diff
@@ -180,74 +182,150 @@ pub trait ConfigServer {
 // consistent with BalanceCache
 pub struct CachedOrgClient {
     pub keypair: Keypair,
-    pub enabled_clients: HashMap<u64, bool>,
     pub client: OrgClient<Channel>,
 }
 
 impl CachedOrgClient {
-    pub fn new(client: OrgClient<Channel>, keypair: Keypair) -> Self {
-        CachedOrgClient {
-            keypair,
-            enabled_clients: HashMap::new(),
-            client,
+    pub fn new(client: OrgClient<Channel>, keypair: Keypair) -> Arc<Mutex<Self>> {
+        Arc::new(Mutex::new(CachedOrgClient { keypair, client }))
+    }
+
+    async fn enable_org(&mut self, oui: u64) -> Result<(), OrgClientError> {
+        tracing::info!(%oui, "enabling org");
+
+        let mut req = OrgEnableReqV1 {
+            oui,
+            timestamp: Utc::now().timestamp_millis() as u64,
+            signer: self.keypair.public_key().into(),
+            signature: vec![],
+        };
+        let signature = self.keypair.sign(&req.encode_to_vec())?;
+        req.signature = signature;
+        let _ = self.client.enable(req).await?;
+        Ok(())
+    }
+
+    async fn disable_org(&mut self, oui: u64) -> Result<(), OrgClientError> {
+        tracing::info!(%oui, "disabling org");
+
+        let mut req = OrgDisableReqV1 {
+            oui,
+            timestamp: Utc::now().timestamp_millis() as u64,
+            signer: self.keypair.public_key().into(),
+            signature: vec![],
+        };
+        let signature = self.keypair.sign(&req.encode_to_vec())?;
+        req.signature = signature;
+        let _ = self.client.disable(req).await?;
+        Ok(())
+    }
+
+    pub fn monitor_funds<S>(
+        client: Arc<Mutex<Self>>,
+        solana: S,
+        minimum_allowed_balance: u64,
+        monitor_period: Duration,
+        shutdown: triggered::Listener,
+    ) -> impl std::future::Future<Output = Result<(), MonitorError<S::Error>>>
+    where
+        S: SolanaNetwork,
+    {
+        let join_handle = tokio::spawn(async move {
+            // TODO: Move this somewhere else, use retainer
+            let mut org_cache = HashMap::new();
+            loop {
+                tracing::info!("Checking if any orgs need to be re-enabled");
+
+                // Fetch all disables orgs:
+                let mut disabled_clients = HashSet::new();
+                let orgs = client
+                    .lock()
+                    .await
+                    .client
+                    .list(OrgListReqV1 {})
+                    .await
+                    .map_err(OrgClientError::RpcError)?
+                    .into_inner();
+                for org in orgs.orgs {
+                    if org.locked {
+                        disabled_clients.insert(org.oui);
+                        let payer = client.fetch_org(org.oui, &mut org_cache).await?;
```

Comment:
i don't see you using `disabled_clients` anywhere, you're just inserting into it and then dropping it at the end of each loop?
also, i don't think you need the outer `org_cache` any longer since you're calling `org list` on each run and that's sending you back the org record, including the `payer` field you're using to query solana. unlike the other, you _are_ using this cache from one run to the next, but since you're always calling the `org list` command and getting back the same info you need the cache isn't doing anything for you